### PR TITLE
increase test coverage

### PR DESCRIPTION
**Pull request by poneill**
Add some more tests, including fuzzing of the API.

Fuzzing revealed that _decode_beams_mp_safe was missing a required arg, so patch that through as well, adding DEFAULT_PRUNE_HISTORY to constants.

### CODE REVIEW

File: pyctcdecode/decoder.py [hunk 1]
```diff
@@ -579,6 +581,7 @@ def decode_beams_batch(
             beam_prune_logp=beam_prune_logp,
             token_min_logp=token_min_logp,
             hotwords=hotwords,
+            prune_history=prune_history,
```
Comment (poneill) [inline pyctcdecode/decoder.py:584]: This seems to have been a bug.

File: pyctcdecode/decoder.py [hunk 2]
```diff
@@ -556,6 +557,7 @@ def decode_beams_batch(
         beam_width: int = DEFAULT_BEAM_WIDTH,
         beam_prune_logp: float = DEFAULT_PRUNE_LOGP,
         token_min_logp: float = DEFAULT_MIN_TOKEN_LOGP,
+        prune_history: bool = DEFAULT_PRUNE_BEAMS,
```
Comment (gkucsko) [inline pyctcdecode/decoder.py:560]: probably add the same constant in `decode_beams`?

**Review by gkucsko — COMMENTED** [review]
nice addition tyty, can you just make triple sure around test runtime and flakiness? very traumatized from using hypothesis

**Comment by poneill** [comment]
yeah for sure, doing that now.

**Review by gkucsko — APPROVED** [review]
lgtm

**Pull request merged**